api/src/database/migrations/20201105A-add-cascade-system-relations.ts

```diff
@@ -132,8 +133,8 @@ export async function up(knex: Knex) {
 				table
 					.foreign(constraint.column)
 					.references(constraint.references)
-					.onUpdate('CASCADE')
-					.onDelete(constraint.onDelete);
+					.onUpdate('NO ACTION')
+					.onDelete('NO ACTION');
```

Comment:
This breaks a lot of other things in the other databases 😬

Reply:
It does, but not on self-referencing tables

Reply:
It's also changing a migration. This should probably be in a conditional step since other databases should still run cascade instead.

Reply:
yeah, that fix was a bit brutal. :) Definitely like the idea of skipping the ON CASCADE of driver is MSSQL

Reply:
I'll look how I can implement the same CASCADE behavior via a trigger

Reply:
done, I've limited the "NO ACTION" only for MSSQL. Unfortunately circular references are created also between different tables so I need to look at this a bit more closely. Should be ok for now, anyway. I'll open an issue to keep track of this